packages/compiler/src/render3/view/api.ts

```diff
@@ -119,6 +119,50 @@ export interface R3DirectiveMetadata {
   providers: o.Expression|null;
 }
 
+/**
+ * Specifies how a list of declaration type references should be emitted into the generated code.
+ */
+export const enum DeclarationEmitMode {
```

Comment:
Could we make this a bit more explicit?

```suggestion
export const enum DeclarationListEmitMode {
```

And similarly for the variable and property names?

Reply:
Done